toplevel/native/topeval.ml

```diff
@@ -181,15 +181,17 @@ let execute_phrase print_outcome ppf phr =
          bytecode and native toplevels always type-check _exactly_ the same
          expression. Adding the binding at the parsetree level (before typing)
          can create observable differences (e.g. in type variable names, see
-         tool-toplevel/pr10712.ml in the testsuite) *)
+         tool-toplevel/topeval.ml in the testsuite) *)
       let str, sg', rewritten =
          match str.str_items with
          | [ { str_desc = Tstr_eval (e, attrs) ; str_loc = loc } ]
          | [ { str_desc = Tstr_value (Asttypes.Nonrecursive,
                                        [{ vb_expr = e
                                         ; vb_pat =
-                                            { pat_desc = Tpat_any;
-                                              pat_extra = []; _ }
+                                            { pat_desc = Tpat_any
+                                            ; pat_extra =
+                                                ([] | [Tpat_constraint _, _, _])
```

Comment:
Is it "safe" to ignore `pat_extra` completely? That then admits at least:
- `let List.(_) = 42` (and also the more complicated `let List.(String.(_)) = ...` etc.)
- `let (_ : int) : int = 42`
- `let List.(_) : int = 42`